### PR TITLE
Allow showing field on ui even if it is set as never_persist 

### PR DESCRIPTION
**Pull request by gowrav-vishwakarma**
if never_persist is set, and ui->editable is set true, even then Field is shown as non-editable. This correction is done to fix the issue.

### CODE REVIEW

**Comment by gowrav-vishwakarma** [comment]
Use 
`$model->addField('description',['never_persist'=>true,'ui'=>['editable'=>true]]);`

**Comment by gowrav-vishwakarma** [comment]
Does this pull request sound okay?

As without it, there are two problems

1. No use of never_persist fields as they are disabled in forms no matter if you set them as ui->editbale->true
2. You don't get these never_persist fields in various hooks, so it makes them of no use.

File: src/Model.php [hunk 1]
```diff
@@ -1442,54 +1442,61 @@ public function save($data = [], Persistence $to_persistence = null)
...
                     }
                 }
 
-                // No save needed, nothing was changed
-                if (!$data && !$dirty_join) {
-                    return $this;
-                }
 
-                if ($this->hook('beforeUpdate', [&$data]) === false) {
+                if ($this->hook('beforeUpdate', [&$data_all]) === false) {
```
Comment (romaninsh) [inline src/Model.php]: $val = $m['field_never_persisted'];

File: src/Field.php [hunk 2]
```diff
@@ -439,7 +439,7 @@ public function useAlias()
      */
     public function isEditable()
     {
-        return $this->read_only || $this->never_persist
+        return !$this->ui['editable'] && ($this->read_only || $this->never_persist)
```
Comment (DarkSide666) [inline src/Field.php]: Probably this condition could be simplified to not check ui/editable multiple times:
```
return isset($this->ui['editable]) ? $this->ui['editable']
    : !( $this->read_only || $this->never_persist || $this->system );
```

**Comment by gowrav-vishwakarma** [comment]
Can be better way to do this

**Pull request closed without merging**